CONTRIBUTING.rst

```diff
@@ -0,0 +1,500 @@
+   Jira issue, we suggest including the Jira issue number in the branch name)::
+
+     $ git checkout master
+     $ git checkout -b fix_comment_typo
+
+   Some Zephyr subsystems do development work on a separate branch from
+   master so you may need to indidicate this in your checkout::
+
+     $ git checkout -b fix_out_of_date_patch origin/net
+
+#. Make changes, test locally, change, test, test again, ...  (Check out the
+   prior chapter on `sanitycheck`_ as well).
+
+#. When things look good, start the pull-request process by adding your changed
+   files::
+
+     $ git add [file(s) that changed, add -p if you want to be more specific]
+
+   You can see files that are not yet staged using::
+
+     $ git status
+
+#. Verify changes to be committed look as you expected::
+
+     $ git diff --cached
+
+#. Commit your changes to your local repo::
+
+     $ git commit -s
+
+   The ``-s`` option automatically adds your ``Signed-off-by:`` to your commit
+   message.  Your commit will be rejected without this line that indicates your
+   agreement with the `DCO`_.  See the `Commit Guidelines`_ section for
+   specific guidelines for writing your commit messages.
+
+
+#. Push your topic branch with your changes to your fork in your personal
+   GitHub account::
+
+     $ git push origin fix_comment_typo
+
+   or if working on a different branch than ``master`` (as mentioned above)::
+
+     $ git push origin/net fix_out_of_date_patch
+
+#. In your web browser, go to your forked repo and click on the
+   ``Compare & pull request`` button for the branch you just worked on and
+   you want to open a pull request with.
+
+#. Review the pull request changes, and verify that you are opening a
+   pull request for the appropriate branch. The title and message from your
+   commit message should appear as well.
+
+#. If you're working on a subsystem branch that's not ``master``,
+   you may need to change the intended branch for the pull-request
+   here, for example, by changing the base branch from ``master`` to ``net``.
+
+#. GitHub will assign one or more suggested reviewers (based on the
+   CODEOWNERS file in the repo) and you can also select additional reviewers
+   now too.
+
+#. Click on the submit button and your pull-request is sent and awaits
+   review.  Email will be sent as review comments are made, or you can check
+   on your pull-request at https://github.com/zephyrproject-rtos/zephyr/pulls.
+
+#. While you're waiting for your pull-request to be accepted and merged, you
+   can create another branch to work on another issue. (Be sure to make your
+   new branch off of master and not the previous branch.)::
+
+     $ git checkout master
+     $ git checkout -b fix_another_issue
+
+   and use the same process described above to work on this new topic branch.
+
+#. If reviewers do request changes to your patch, go back to your local repo,
+   verify you're on the right branch, make changes, ammend your commit, rebase
```

Comment:
s/ammend/amend/

Here again the workflow assumes the PR has only one commit. Suggest folding the recommendations to use interactive rebase into  the main workflow sequence instead.

The `--autosquash` option to `git rebase -i` is particularly useful for handling review feedback of the form "you forgot to add a space here"; maybe worth a mention.